src/net/p2p.rs

```diff
@@ -3986,6 +4029,88 @@ impl PeerNetwork {
         Ok(())
     }
 
+    /// Set up the unconfirmed chain state off of the canonical chain tip.
+    fn setup_unconfirmed_state(
+        chainstate: &mut StacksChainState,
+        sortdb: &SortitionDB,
+    ) -> Result<(), Error> {
+        let (canonical_consensus_hash, canonical_block_hash) =
+            SortitionDB::get_canonical_stacks_chain_tip_hash(sortdb.conn())?;
+        let canonical_tip = StacksBlockHeader::make_index_block_hash(
+            &canonical_consensus_hash,
+            &canonical_block_hash,
+        );
+        // setup unconfirmed state off of this tip
+        debug!(
+            "Reload unconfirmed state off of {}/{}",
+            &canonical_consensus_hash, &canonical_block_hash
+        );
+        chainstate.reload_unconfirmed_state(&sortdb.index_conn(), canonical_tip)?;
+        Ok(())
+    }
+
+    /// Store a single transaction
+    /// Return true if stored; false if it was a dup.
+    /// Has to be done here, since only the p2p network has the unconfirmed state.
+    fn store_transaction(
+        mempool: &mut MemPoolDB,
+        chainstate: &mut StacksChainState,
+        consensus_hash: &ConsensusHash,
+        block_hash: &BlockHeaderHash,
+        tx: StacksTransaction,
+    ) -> bool {
+        let txid = tx.txid();
+        if mempool.has_tx(&txid) {
+            debug!("Already have tx {}", txid);
+            return false;
+        }
+
+        if let Err(e) = mempool.submit(chainstate, consensus_hash, block_hash, tx) {
+            info!("Reject transaction {}: {:?}", txid, &e);
```

Comment:
Can you use the key-value format for `info` logs? It'll make ingesting logs a lot easier for different frameworks (not just kibana, but our testground harness also will process JSON logs).

Reply:
Addressed in c73b3f5